lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -405,7 +405,7 @@ <h4 class="message-title">${_('Your current verification will expire soon.')}</h
               </p>
               <div class="action-upgrade-container">
                 % if use_ecommerce_payment_flow and course_mode_info['verified_sku']:
-                  <a class="action action-upgrade" href="${ecommerce_payment_page}?sku=${course_mode_info['verified_sku']}">
+                  <a id="ecommerce_upgrade_link" class="action action-upgrade" href="${ecommerce_payment_page}?sku=${course_mode_info['verified_sku']}">
```

Comment:
It looks like you're binding the javascript to the link click via the id value 'ecommerce_upgrade_link', is that right? Some pages will pull in multiple upsell links- so this name would be a unique for the specific section/banner its in, but not a unique id for the DOM- does that make trouble? Could it be safer to bind to something more general, like class?

Reply:
Yes, I am binding the event to this object by id, which should be unique per element in the page. I only intend for the event to be attached to this link for the scope of this ticket, which is why I'm using the ID. 

If there are multiple links/buttons on one page, we'd still have to somehow identify which one is which (since we've named each link/button uniquely). Whether that's done via ID or some other mechanism we should probably figure out and make changes accordingly in a separate PR.

Reply:
barring a way of identifying this on a per-instance-on-page basis, I think binding by ID is reasonable, but would want to verify on a page with multiple that you don't get repeat submissions.

Reply:
Cool- we know we'll have pages with multiple upsell links, so we we can switch to class="ecommerce_upgrade_link" (the current upsell event code does this successfully with a different class name)